KIPs/kip-146.md

```diff
@@ -0,0 +1,96 @@
+---
+kip: 146
+title: Random Proposer Selection
+author: Ian (@ian0371), Ollie (@blukat29), Joseph (@jiseongnoh), and Aidan (@aidan-kwon)
+discussions-to: https://github.com/klaytn/kips/issues/146
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-06-15
+---
+
+## Simple Summary
+
+An unpredictable and verifiable proposer selection policy
+
+## Abstract
+
+This standard outlines an approach for a random proposer selection policy. A proposer is determined every block in a random manner. The randomness relies on [KIP-114](https://github.com/klaytn/kips/blob/main/KIPs/kip-114.md), thereby rendering the new policy both unpredictable and verifiable.
+
+## Motivation
+
+Klaytn has adopted the Byzantine Fault Tolerance (BFT) consensus mechanism, where a block proposer selected by a deterministic algorithm proposes the next block. A vulnerable point of BFT is that an adversary can effectively halt the entire network by simply targeting the proposer, a.k.a. targeted DoS. What makes Klaytn even more vulnerable to a targeted DoS attack is that the current proposer selection policy enables the prediction of up to `proposerRefreshInterval` (3600 in case of Cypress) number of proposers in advance.
+
+As of now, a targeted DoS is infeasible in Klaytn because it is a permissioned chain which allows only authorized validators to join the network. However, in permissionless network, a targeted DoS is a viable option for attackers. A random proposer selection can increase the difficulty of the attack by introducing an uncertainty of the upcoming proposers.
```

Comment:
> Finding out the IP addresses of CNs (Consensus Node) is difficult, because you need to become a GC to find out other CN's IP addresses.

So, I taken a simple one possible example (malicious CN)

> Even if you find out the IP address, CN is run behind a firewall that only allows other CNs.

It depends on the attacker's limitation and strategy. Neither provided by Klaytn architecture nor implementation.

Thus, the conclusion was 

> it seems that the attack is logically not infeasible.

Hopefully, the term "practically infeasible" effectively conveys the intended meaning to discerning readers.